frontend/gpay_agent/lib/main.dart

```diff
@@ -16,7 +18,56 @@ class AgentApp extends StatelessWidget {
           primarySwatch: Colors.blue,
           visualDensity: VisualDensity.adaptivePlatformDensity,
         ),
-        home: LoginView(),
+        home: StartAgentApp(),
     );
   }
-}
\ No newline at end of file
+}
+
+///Builds [StartAgentAppState]
+class StartAgentApp extends StatefulWidget{
+  @override
+  State<StatefulWidget> createState() {
+    return StartAgentAppState();
+  }
+}
+
+///Starts capturing GPS location of device. Redirects to [LoginView]
+class StartAgentAppState extends State<StartAgentApp>{
+  Geolocator _geolocator;
+  Position _position;
+
+  ///Checks the status of GPS granted by user
+  ///locationAlways means GPS location will be traced when app is running either in background or foreground.
+  ///locationWhenInUse means GPS location will be traced when this application is the running app.
+  ///Default status will be granted if either of the two is granted
+  void checkPermission() {
+    _geolocator.checkGeolocationPermissionStatus().then((status) { print('status: $status'); });
+    _geolocator.checkGeolocationPermissionStatus(locationPermission: GeolocationPermission.locationAlways).then((status) { print('always status: $status'); });
+    _geolocator.checkGeolocationPermissionStatus(locationPermission: GeolocationPermission.locationWhenInUse)..then((status) { print('whenInUse status: $status'); });
+  }
+
+  ///Initializes _goelocator and stream for listening changes in device location.
+  ///Position will be changed if distance changes by [distanceFilter]
+  @override
+  void initState() {
+    super.initState();
+
+    _geolocator = Geolocator();
+    LocationOptions locationOptions = LocationOptions(accuracy: LocationAccuracy.high, distanceFilter: 1);
```

Comment:
I would recommend storing this in a constant int member of class like UPDATE_DISTANCE